src/node_native_module_env.cc

```diff
@@ -0,0 +1,203 @@
+#include "node_native_module_env.h"
+#include "env-inl.h"
+
+namespace node {
+namespace native_module {
+
+using v8::Context;
+using v8::DEFAULT;
+using v8::Function;
+using v8::FunctionCallbackInfo;
+using v8::IntegrityLevel;
+using v8::Isolate;
+using v8::Local;
+using v8::Maybe;
+using v8::MaybeLocal;
+using v8::Name;
+using v8::None;
+using v8::Object;
+using v8::PropertyCallbackInfo;
+using v8::Set;
+using v8::SideEffectType;
+using v8::String;
+using v8::Value;
+
+// TODO(joyeecheung): make these more general and put them into util.h
+Local<Set> ToJsSet(Local<Context> context, const std::set<std::string>& in) {
+  Isolate* isolate = context->GetIsolate();
+  Local<Set> out = Set::New(isolate);
+  for (auto const& x : in) {
+    out->Add(context, OneByteString(isolate, x.c_str(), x.size()))
+        .ToLocalChecked();
+  }
+  return out;
+}
+
+bool NativeModuleEnv::Exists(const char* id) {
```

Comment:
IMO this is a bit of an abstraction break. I think it should be replaced with `HasThirdPartyMain` based on knowledge known at compile time

Reply:
I can make specialize it into a run time `HasThirdPartyMain`, but I would prefer making this compile-time in another patch.

(You can still patch this to include a `_third_party_main.js` at run time, which I think is useful considering now if you don't insert this Node would straight goes into the REPL when embedded)

Reply:
Ack.